lib/open_uri.ml

```diff
@@ -15,22 +15,24 @@
  *
  *)
 
-let handle_socket f s =
-  try
-    let result = f s in
-    Unix.close s;
-    result
-  with e ->
-    Backtrace.is_important e;
-    Unix.close s;
-    raise e
+let with_socket s sockaddr f =
+  let result =
+    try
+      Unix.connect s sockaddr;
+      f s
+    with e ->
+      Backtrace.is_important e;
+      Unix.close s;
+      raise e
+  in
+  Unix.close s;
```

Comment:
Move `Unix.close s` outside the `try...with` block to avoid double close the socket.